Drivers/Periphs/Src/SDCard.c

```diff
@@ -151,87 +206,67 @@ FRESULT SDCard_OpenFileSystem(){
     return f_mount(&FatFs, "", 1); //1=mount now
 }
 
+/**
+ * @brief Fetch number of dropped SD Card messages due to queue overfilling.
+ *        Included for debug purposes
+ * @return Number of dropped messages
+ */
+uint32_t SDCard_FetchDroppedMsgCnt() {
+    return SDCDroppedMessages;
+}
+
 //STATIC FUNCTIONS FOR WRITING TO SD CARD WITH CORRECT FORMAT
 // Wrappers for snprintf() for each of the message data types
 // Each function takes a buffer to write to, the buffer size, 
 // and a pointer to the respective data structs
 
 static int SPrint_CAN(char *sdcard_write_buf, 
                       size_t bufsize, 
-                      CANMSG_t *can, 
+                      void *can, 
                       const char *time) {
     return snprintf(sdcard_write_buf, 
                     bufsize, 
-                    "%.9s,%d,%d,%lu\r\n", 
+                    "%.9s,%.3" PRIx16 ",%" PRIu8 ",%" PRIx64 "\n", 
                     time,
-                    can->id, 
-                    can->payload.idx, 
-                    can->payload.data.w);
+                    ((CANMSG_t *)can)->id, 
+                    ((CANMSG_t *)can)->payload.idx, 
+                    *(uint64_t *)(((CANMSG_t *)can)->payload.data.bytes));
 }
 
 static int SPrint_IMU(char *sdcard_write_buf, 
                       size_t bufsize, 
-                      IMUData_t *imu, 
+                      void *imu, 
```

Comment:
Not sure why you did this since this function will only be used for the IMU data types...

Reply:
Lets me point to the function in a struct, removing the switch case.